### PR TITLE
Add Shapefile automatically to map

### LINKED ISSUE #702: Add Shapefile automatically to map
When adding SHP, the layer is stored in Layman, but is not added automatically to the map.

### PR DESCRIPTION
**Pull request by jmacura**
fixes #702

### CODE REVIEW

**Comment by jmacura** [comment]
Ugly one. I hate these Layman and Micka related bugs, they are so hard to catch... 🙁

**Comment by raitisbe** [comment]
Maybe in that case it would be better to use the name returned from layman post request to make it futureproof? 
image
Did you notice post requests failing because of bad names?

File: components/add-layers/shp/add-layers-shp.component.ts [hunk 1]
```diff
@@ -69,27 +72,30 @@ export class HsAddLayersShpComponent implements OnInit {
...
+    layerName: string
+  ): Promise<HsLaymanLayerDescriptor> {
+    try {
+      const descriptor = await this.hsLaymanService.describeLayer(
+        endpoint,
+        layerName
+      );
+      if (['STARTED', 'PENDING', 'SUCCESS'].includes(descriptor.wms.status)) {
+        setTimeout(async () => {
+          return await this.describeNewLayer(endpoint, layerName);
```
Comment (raitisbe) [inline components/add-layers/shp/add-layers-shp.component.ts]: I was struggling with this timeout and thats the reason why I havent rewriotten to promises. It was returning inside setTimeout instead of describeNewLayer. I suspect it might be that here too. The way to check would be to upload a large shp file which takes time to process.

Comment (jmacura) [reply components/add-layers/shp/add-layers-shp.component.ts]: You are commenting on an outdated file, are you aware of that? Exactly this part was the problem – I had to promisify the `setTimeout()` with `new Promise()` and resolve from inside of it.

**Comment by jmacura** [comment]
> 
> Maybe in that case it would be better to use the name returned from layman post request to make it futureproof?

Definitely! That way we can go away with the unidecode as well.

**Comment by jmacura** [comment]
> Did you notice post requests failing because of bad names?

I did not. Can you share more details about those failing requests?

**Comment by jmacura** [comment]
[Release 2.5.0](https://trello.com/c/ve259N2U/132-release-250)

**Review by raitisbe — APPROVED** [review]

**Comment by jmacura** [comment]
For backport to 1.x refer #1236 , port to 3.x is needed.
